### PR TITLE
Adds block size metric

### PR DESCRIPTION
**Pull request by eadventurous**
### Description of the Change
Adds block size metric.

### Benefits
This way for projects building on Iroha2 it will be easier to estimate the needed disk space.

### Jira Issue
https://jira.hyperledger.org/browse/IR-1163

### CODE REVIEW

File: iroha/src/kura.rs [hunk 1]
```diff
@@ -404,4 +407,46 @@ mod tests {
...
+                    .unwrap();
+            }
+            let block = block.commit();
+            let block_store = BlockStore::new(dir.path()).unwrap();
+            block_store.write(&block).await.unwrap();
+            let metadata = fs::metadata(block_store.get_block_path(1)).await.unwrap();
+            let file_size = Byte::from_bytes(metadata.len() as u128).get_appropriate_unit(false);
+            println!("For {} validators: {}", n_validators, file_size);
+        }
+    }
```
Comment (i1i1) [inline iroha/src/kura.rs:451]: Technically this is not a test, but a benchmark. May be you can move it accordinally?

File: iroha/src/kura.rs [hunk 2]
```diff
@@ -404,4 +407,46 @@ mod tests {
...
+                    .sign(&KeyPair::generate().expect("Failed to generate KeyPair."))
+                    .unwrap();
+            }
+            let block = block.commit();
+            let block_store = BlockStore::new(dir.path()).unwrap();
+            block_store.write(&block).await.unwrap();
+            let metadata = fs::metadata(block_store.get_block_path(1)).await.unwrap();
+            let file_size = Byte::from_bytes(metadata.len() as u128).get_appropriate_unit(false);
+            println!("For {} validators: {}", n_validators, file_size);
+        }
```
Comment (i1i1) [inline iroha/src/kura.rs:450]: Also may be you can move loop body into some function?

File: iroha/src/kura.rs (hunk 1 above)
Comment (eadventurous) [reply iroha/src/kura.rs:451]: Sadly benchmarks are only focused on measuring speed, so it makes no sense to put it there for it to be ran thousands of times to measure and output speed.

File: iroha/src/kura.rs (hunk 2 above)
Comment (eadventurous) [reply iroha/src/kura.rs:450]: I don't think it makes sense for this simple test. Nothing is shared with the other tests.

**Review by Revertron — DISMISSED** [review]

File: iroha/src/kura.rs (hunk 1 above)
Comment (i1i1) [reply iroha/src/kura.rs:451]: Yeah, but you can run it once and print output. Or you can move it somewhere else. This thing doesnt test anything. I think it is more of a bench than a test :)

File: iroha/src/kura.rs (hunk 2 above)
Comment (i1i1) [reply iroha/src/kura.rs:450]: Yeah, but it would give more context and will be more readable

**Review by rkharisov — APPROVED** [review]

**Review by i1i1 — APPROVED** [review]

File: iroha/benches/kura.rs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    let mut block = PendingBlock::new(vec![tx])
+        .chain_first()
+        .validate(&WorldStateView::new(World::new()), &AllowAll.into());
+    for _ in 0..n_validators {
+        block = block
+            .sign(&KeyPair::generate().expect("Failed to generate KeyPair."))
+            .unwrap();
+    }
+    let block = block.commit();
+    let block_store = BlockStore::new(dir.path()).unwrap();
```
Comment (Revertron) [inline iroha/benches/kura.rs:38]: Could it be implemented without storage, just using serialization?

Comment (eadventurous) [reply iroha/benches/kura.rs:38]: In general it could, but in this case we want to see the actual file size on disk.

Comment (Revertron) [reply iroha/benches/kura.rs:38]: But serialization will give you some vector with exact byte count, no?

**Pull request merged**